### DISCUSSION TITLE
Values are not being audited when using DB::beginTransaction

### DISCUSSION THREAD

**Discussion by alexfarrugia** [owen-it/laravel-auditing#913] [Q&A] [answered]
| Q                 | A
| ----------------- | ---
| Bug?              | maybe
| New Feature?      | no
| Framework         | Laravel
| Framework version | 9.0.0
| Package version   | 12.0.0
| PHP version       | 8.1

#### Actual Behaviour

I am making changes to some models which use the Auditable trait within the context of a db transaction. I need to do this because there are chunk of changes which are happening and i need to be able to rollback if something goes wrong. I have noticed that the changes to the values of these models are being audited as empty objects. (see  example in this image)
![image](https://github.com/owen-it/laravel-auditing/assets/10478855/5e4b9ab2-3ce5-43c6-b71f-2965ec5be333)

Could it be that the audits are happening before the changes are being committed to the database or maybe am i missing something myself?

#### Expected Behaviour

I would expect that the correct new values are being audited in the database.

#### Steps to Reproduce

`DB::beginTransaction();
        try {
      $fund->amount = 5000;
    $fund->save();
 .....

DB::commit();
}`

#### Possible Solutions

Not sure if there is a way of delaying the audit until the actual transaction is committed perhaps ? I can't think of any other way to somehow include the audit within the same database transaction?

thanks

**Comment by alexfarrugia** [comment]
i noticed that this was actually happening when i was using update statements with DB::raw - maybe this could be the reason and not the usage of the DB transaction command?

```
$payment->update([
                            'amount' => DB::raw(amount - allocated_amount),
                        ]);
```

**Comment by parallels999** [comment, marked as answer]
I'm using `DB::beginTransaction` and `DB::raw` the same, and everything works as expected
Only if it a bulk update, in that case it doesn't work
```php
$payment=PaymentModel::query();
$payment->update([
    'amount' => DB::raw('amount - allocated_amount'),
]);
// this is not audited because is a bulk update, and it doesn't trigger laravel update event
```
